### PR TITLE
fix: the random month showing issue in datepicker widget

### LINKED ISSUE #13389: [Bug]-[140]: Date picker shows random month when a custom min date is set
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When we choose a date other than the current date as minDate in datepicker widget, it shows a random month in the date picker popup.
[LOOM DEMO](https://www.loom.com/share/48be010e55284b23b7d59764104247f1)

### Steps To Reproduce

1. Add a Date picker widget to canvas
2. Set tomorrow's date as minDate
3. Remove defaultDate
4. Open the date picker popup on the canvas
5. Observer that it shows a random month 

### Public Sample App

_No response_

### Version

cloud

### PR DESCRIPTION
**Pull request by techbhavin**
## Description

> fix issue of Date picker shows random month when a custom min date is set

Fixes #13389

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/widgets/DatePickerWidget2/component/index.tsx [hunk 1]
```diff
@@ -164,6 +164,15 @@ class DatePickerComponent extends React.Component<
       isValid && this.state.selectedDate
         ? new Date(this.state.selectedDate)
         : null;
+    const initialMonth = this.props.minDate
```
Comment (jsartisan) [inline app/client/src/widgets/DatePickerWidget2/component/index.tsx:167]: This whole block is so hard to read. Can we use a switch statement for this?

**Review by jsartisan — APPROVED** [review]

**Comment by chandannkumar** [comment]
**Observation**:
When we add a MAX date as Older dates, the behavior in date picker is weird:
1. The 'January' Month dates are disabled for 1st time on datepicker popup when MAX date year is set as 2022
2. When we add MAX date as less than year 2022, the popup still shows Latest month & year calendar
2. Since we added MAX date as older date, therefore 'Today' option should also be disabled from popup

https://www.loom.com/share/78d3971c1d504adb832d069fba524251
@techbhavin cc: @dilippitchika

**Comment by techbhavin** [comment]
> **Observation**: When we add a MAX date as Older dates, the behavior in date picker is weird:
> 
> 1. The 'January' Month dates are disabled for 1st time on datepicker popup when MAX date year is set as 2022
> 2. When we add MAX date as less than year 2022, the popup still shows Latest month & year calendar
> 3. Since we added MAX date as older date, therefore 'Today' option should also be disabled from popup
> 
> https://www.loom.com/share/78d3971c1d504adb832d069fba524251 @techbhavin cc: @dilippitchika

We can not disable the "Today" button and also do not control this button from us. We can just hide both footer buttons.
@chandannkumar 
cc: @dilippitchika

**Comment by chandannkumar** [comment]
Tested this PR and working as expected.
Tested for MIN & MAX date behavior on datepicker popup

File: app/client/src/widgets/DatePickerWidget2/component/index.tsx [hunk 2]
```diff
@@ -164,6 +164,90 @@ class DatePickerComponent extends React.Component<
       isValid && this.state.selectedDate
         ? new Date(this.state.selectedDate)
         : null;
+
+    const getInitialMonth = () => {
```
Comment (jsartisan) [inline app/client/src/widgets/DatePickerWidget2/component/index.tsx:168]: I think we might need useMemo here.

**Review by jsartisan — APPROVED** [review]

**Review by jsartisan — APPROVED** [review]

**Pull request merged**